paimon-common/src/main/java/org/apache/paimon/utils/FileBasedBloomFilter.java

```diff
@@ -34,16 +35,18 @@
 /** Util to apply a built bloom filter . */
 public class FileBasedBloomFilter implements Closeable {
 
-    private final PageFileInput input;
+    private final SeekableInputStream input;
     private final CacheManager cacheManager;
     private final BloomFilter filter;
     private final long readOffset;
-    private final int readLength;
     private final CacheKey cacheKey;
+    // each bloom filter is only used by a single file reader, so we can safely reuse here
+    private final byte[] reusedPageBuffer;
```

Comment:
Do not introduce this, when memory preemption occurs, it's precisely when memory usage needs to be cleaned up. Your way of referencing it makes it impossible to clean up.